python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -1,187 +1,210 @@
-        # back on the full-res version and will succeed. This handles that case
-        # and any custom cases where you may want to send your proxy render to 
-        # screening room. 
-        root_node = nuke.root()
-        is_proxy = root_node['proxy'].value()
-        if is_proxy:
-            self.__app.log_info("Proxy mode is ON. Rendering proxy.")
-            node["proxy"].setValue(path.replace(os.sep, "/"))
+        writenode_quicktime_settings = self.__app.execute_hook_method("codec_settings_hook",
+                                                                      "get_quicktime_settings")
+
+        render_script_path = os.path.join(self.__app.disk_location, "hooks",
+                                          "nuke_batch_render_movie.py")
+        ctx = self.__app.context
+
+        shotgun_context = {}
+        shotgun_context['entity'] = ctx.entity.copy()
+        if ctx.task:
+            shotgun_context['task'] = ctx.task.copy()
+        if ctx.step:
+            shotgun_context['step'] = ctx.step.copy()
+        shotgun_context['project'] = ctx.project.copy()
+
+        app_settings = {
+            'version_number_padding': self.__app.get_setting('version_number_padding'),
+            'slate_logo': self._logo,
+        }
+
+        render_info = {
+            'burnin_nk': self._burnin_nk,
+            'slate_font': self._font,
+            'codec_settings': {'quicktime': writenode_quicktime_settings},
+        }
+
+        # set needed paths and force them to use forward slashes for use in Nuke (latter needed for Windows)
+        src_frames_path = path.replace('\\', '/')
+        movie_output_path = output_path.replace('\\', '/')
+
+        nuke_render_info = {
+            'width': width,
+            'height': height,
+            'first_frame': first_frame,
+            'last_frame': last_frame,
+            'version': version,
+            'name': name,
+            'color_space': color_space,
+            'nuke_exe_path': nuke_exe_path,
+            'render_script_path': render_script_path,
+            'shotgun_context': shotgun_context,
+            'app_settings': app_settings,
+            'render_info': render_info,
+            'src_frames_path': src_frames_path,
+            'movie_output_path': movie_output_path,
+        }
+
+        return nuke_render_info
+
+    def render_movie_in_nuke(self, path, output_path,
+                             width, height,
+                             first_frame, last_frame,
+                             version, name,
+                             color_space,
+                             active_progress_info=None):
+
+        render_info = self.gather_nuke_render_info(path, output_path, width, height, first_frame,
+                                                   last_frame, version, name, color_space)
+        # TODO: can we offload to a thread, similar to submitter?
+        run_in_batch_mode = True if nuke is None else False
```

Comment:
This is fine for now. Eventually, we will probably want another way to control this, as it is a valid case to want to offline a movie render even when in a Nuke session.